src/tests/optval/test_optval.f90

```diff
@@ -60,6 +63,36 @@ function foo_dp(x) result(z)
   endfunction foo_dp
 
 
+  subroutine test_optval_csp
+    complex(sp) :: z1
+    print *, "test_optval_csp"
+    z1 = cmplx(1.0_sp, 2.0_sp)
```

Comment:
```suggestion
    z1 = cmplx(1.0_sp, 2.0_sp, kind=sp)
```
Nitpick I missed the first time: `cmplx` without an explicit KIND argument produces a default-kind complex value, regardless of the kinds of its X and Y arguments. Here, it turns out not to matter, but we should do the "right" thing anyway.